src/world/Explosion.php

```diff
@@ -118,6 +129,9 @@ public function explodeA() : bool{
 							}
 
 							$state = $subChunk->getBlockStateId($vBlockX & SubChunk::COORD_MASK, $vBlockY & SubChunk::COORD_MASK, $vBlockZ & SubChunk::COORD_MASK);
+							if(isset($this->excludedBlockTypeIds[$blockFactory->fromStateId($state)->getTypeId()])){
```

Comment:
This check can be avoided for the common case by checking if the count of excluded IDs is > 0. Reading stuff from BlockFactory is costly.